plugins/kubernetes/test/models/kubernetes/template_filler_test.rb

```diff
@@ -881,6 +881,20 @@ def secret_annotations(hash)
         template.to_hash[:spec][:foo].must_equal "bar"
       end
 
+      it "sets annotations with ." do
+        raw_template[:metadata][:annotations] = {
+          "samson/set_via_env_json-metadata.annotations.foo.bar/baz" => "FOO"
+        }
+        template.to_hash[:metadata][:annotations][:"foo.bar/baz"].must_equal "bar"
+      end
+
+      it "sets labels with ." do
+        raw_template[:metadata][:annotations] = {
```

Comment:
shouldn't this be under labels instead of annotations?

Reply:
the replacement always gets set as an annotation, but it can replace labels